packages/cli/src/executions/execution.service.ts

```diff
@@ -495,4 +502,26 @@ export class ExecutionService {
 			s.scopes = scopes[s.workflowId] ?? [];
 		}
 	}
+
+	public async annotate(
+		executionId: string,
+		updateData: { tags: string[]; vote: AnnotationVote | null },
+	) {
+		// FIXME: wrap in transaction
+		await this.executionAnnotationRepository.upsert(
+			{ execution: { id: executionId }, vote: updateData.vote },
+			['execution'],
+		);
+
+		if (updateData.tags) {
+			await this.annotationTagMappingRepository.overwriteTags(executionId, updateData.tags);
+		}
+
+		const execution = await this.executionRepository.findSingleExecution(executionId, {
+			includeAnnotation: true,
+			includeData: true,
+		});
```

Comment:
Could we move this away from the `annotate` method so that it only annotates, and the execution fetching is done outside this function to keep this single purpose

Reply:
I moved it to the controller